src/libraries/System.Runtime.Numerics/src/System/Number.BigInteger.cs

```diff
@@ -1437,4 +1309,97 @@ internal static bool TryFormatBigInteger(BigInteger value, ReadOnlySpan<char> fo
+
+        static virtual int DigitsPerBlock => sizeof(uint) * 8 / TParsingInfo.BitsPerDigit;
+
+        static abstract NumberStyles BlockNumberStyle { get; }
+
+        static abstract uint GetSignBitsIfValid(uint ch);
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        static virtual bool TryParseUnalignedBlock(ReadOnlySpan<TChar> input, out uint result)
+        {
+            if (typeof(TChar) == typeof(char))
+            {
+                return uint.TryParse(MemoryMarshal.Cast<TChar, char>(input), TParsingInfo.BlockNumberStyle, null, out result);
+            }
+
+            throw new NotSupportedException();
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        static virtual bool TryParseSingleBlock(ReadOnlySpan<TChar> input, out uint result)
+            => TParsingInfo.TryParseUnalignedBlock(input, out result);
+
+        static virtual bool TryParseWholeBlocks(ReadOnlySpan<TChar> input, Span<uint> destiniation)
+        {
+            Debug.Assert(destiniation.Length * TParsingInfo.DigitsPerBlock == input.Length);
+            ref TChar lastWholeBlockStart = ref Unsafe.Add(ref MemoryMarshal.GetReference(input), input.Length - TParsingInfo.DigitsPerBlock);
+
+            for (int i = 0; i < destiniation.Length - 1; i++)
+            {
+                if (!TParsingInfo.TryParseSingleBlock(
+                    MemoryMarshal.CreateReadOnlySpan(ref Unsafe.Subtract(ref lastWholeBlockStart, i * TParsingInfo.DigitsPerBlock), TParsingInfo.DigitsPerBlock),
+                    out destiniation[i]))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+    }
+
+    internal readonly struct BigIntegerHexParsingInfo<TChar> : IBigIntegerHexOrBinaryParsingInfo<BigIntegerHexParsingInfo<TChar>, TChar>
+        where TChar : unmanaged, IBinaryInteger<TChar>
+    {
+        public static int BitsPerDigit => 4;
+
+        public static NumberStyles BlockNumberStyle => NumberStyles.AllowHexSpecifier;
+
+        // A valid ASCII hex digit is positive (0-7) if it starts with 00110
+        public static uint GetSignBitsIfValid(uint ch) => (uint)((ch & 0b_1111_1000) == 0b_0011_0000 ? 0 : -1);
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool TryParseWholeBlocks(ReadOnlySpan<TChar> input, Span<uint> destiniation)
+        {
+            if (typeof(TChar) == typeof(char))
+            {
+                if (Convert.FromHexString(MemoryMarshal.Cast<TChar, char>(input), MemoryMarshal.AsBytes(destiniation), out _, out _) != OperationStatus.Done)
+                {
+                    return false;
+                }
+
+                if (BitConverter.IsLittleEndian)
+                {
+                    MemoryMarshal.AsBytes(destiniation).Reverse();
+                }
+                else
+                {
+                    destiniation.Reverse();
+                }
```

Comment:
This can be improved if there's a vectorized path that parses in reverse byte order. Performance for huge numbers should be improved, however for the most interesting 96-256bit cases, I'd expect the performance comparison to be very complicated. Thus I'm choosing the simplest approach to depend on public API.

Convert.FromHexString has a slight overhead over HexConverter, but the latter is only vectorized in CoreLib.